### DISCUSSION TITLE
Problemi riavvio modem e hassio

### DISCUSSION THREAD

**Discussion by TheSkull00** [indomus/forum#263] [HUB: Home Assistant]
salve a tutti. Vi spiego subito il problema. Praticamente ho integrato due interruttori sonoff con hassio, tramite un'automazione accendono le luci esterne al tramonto. Il tutto funziona alla grande il problema si presenta quando magari nel corso della giornata per qualche motivo si interrompe la corrente (lavori da parte dell'enel, temporale ecc.) ovviamene il mio raspberry e i miei sonoff si avviano più velocemente rispetto a quanto ci impiega il router a riconnettersi pienamente alla linea. e questo fa si che hassio non rivela i due sonoff e quindi l'automazione non funziona. Nei log ho il messaggio di errore di configurazione non valida per quanto riguarda i sonoff. Lo stesso problema si presenta con alexa, se va via la corrente non riproduce più notifiche da parte di hassio. Il tutto si risolve con un semplice riavvio di hassio ma vorrei sapere da voi qualche soluzione per risolvere il problema in maniera definitiva.

**Comment by essegi65** [comment]
Io per evitare qualsiasi problema ho preso un UPS su internet e ci ho collegato sia il modem che il raspberry con HA (gli assorbimenti di entrambi sono ridicoli, quindi se lo prendi un pò sovradimensionato la durata della batteria è sufficiente). Non solo evito problemi con la caduta di corrente ma posso continuare ad usare anche alcuni componenti di HA. Al momento non ho mai avuto problemi di capacità della batteria dell'ups in quanto la corrente di solito è tornata sempre prima che andasse in esaurimento.

**Reply by TheSkull00** [reply]
Non c'è un modo per riavviare automaticamente hassio dopo che viene rilevata la mancata connessione?

**Reply by snakuzzo** [reply]
1. Gli errori in configurazione non sono normali...ma...domanda...come hai integrato i sonoff ?
2. I problemi non si risolvono riavviando HA

**Reply by TheSkull00** [reply]
Forse mi sono spiegato male. L'errore in configurazione mi esce quando si avvia l'automazione e i sonoff non risultano collegati ad hassio. In condizioni normali non mi escono errori. Praticamente hassio non li rivela perché quando torna la corrente  il Raspberry e tutte le sue integrazioni si avviano più velocemente rispetto al modem. Quindi i sonoff sono accessi e non connessi per due/tre minuti ( il tempo che si avvia il modem). Infatti se riavvio hassio dopo che il modem è completamente attivo rileva tranquillamente i sonoff perché sono già connessi a Internet. 
I sonoff sono integrati tramite sonoffLan con modalità Cloud

**Reply by snakuzzo** [reply]
puoi semplicemente fare un reload delle integrazioni
Lo fai tramite API locali esposte da HA
Stessa cosa puoi farla se vuoi effettuare un reboot. Su Hassio tra le altre cose dovresti avere dei service già pronti

**Comment by 50494554524F** [comment]
lo stato dei sensori sarà undefined o unavailable, metti un controllo su questo e dopo tot minuti usa il comando per riavviare ha
